src/components/MovieDialog/MovieDialog.js

```diff
@@ -1,50 +1,91 @@
 
 const MovieDialog = (props) => {
-  const dialogInfo = {
-    dialogType: props.movie ? "Edit Movie" : "Add Movie",
-    title: props.movie ? props.movie.title : "",
-    releaseDate: props.movie ? props.movie.release_date : "",
-    posterPath: props.movie ? props.movie.poster_path : "",
-    rating: props.movie ? props.movie.vote_average : "",
-    genres: props.movie ? props.movie.genres : "",
-    runtime: props.movie ? props.movie.runtime : "",
-    overview: props.movie ? props.movie.overview : "",
-  };
-  const [dialogData, setDialogData] = useState(dialogInfo);
+  const dispatch = useDispatch();
 
-  const dataChangeHandler = (event) => {
-    const name = event.target.name;
-    setDialogData((prevState) => {
-      return { ...prevState, [name]: event.target.value };
-    });
-  };
   const closeDialog = () => {
     props.setMovieDialogVisibility(false);
   };
 
-  const reset = () => {
-    setDialogData(dialogInfo);
-  };
+  const validate = (values) => {
+    const errors = {};
+    if (!values.title) {
+      errors.title = "Title is required";
+    }
+    if (!values.releaseDate) {
+      errors.releaseDate = "Date is required";
+    }
+    if (!values.posterPath) {
+      errors.posterPath = "Movie URL is required";
+    } else if (
+      !/^(?:(?:https?|ftp):\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,}))\.?)(?::\d{2,5})?(?:[/?#]\S*)?$/i.test(
+        values.posterPath
+      )
+    ) {
+      errors.posterPath = "Invalid Movie URL";
+    }
+
+    if (!values.rating) {
+      errors.rating = "Rating is required";
+    }
+
+    if (!values.overview) {
+      errors.overview = "Overview is required";
+    }
 
-  const submitHandler = (event) => {
-    event.preventDefault();
-    console.log(dialogData);
+    if (!values.runtime) {
+      errors.runtime = "Runtime is required";
+    }
+
+    if (!values.genres.length) {
+      errors.genres = "Genre is required";
+    }
+
+    return errors;
   };
 
+  const formik = useFormik({
+    initialValues: {
+      dialogType: props.movie ? "Edit Movie" : "Add Movie",
+      title: props.movie ? props.movie.title : "",
+      releaseDate: props.movie ? props.movie.release_date : "",
+      posterPath: props.movie ? props.movie.poster_path : "",
+      rating: props.movie ? props.movie.vote_average : "",
+      genres: props.movie ? props.movie.genres : [],
+      runtime: props.movie ? props.movie.runtime : "",
+      overview: props.movie ? props.movie.overview : "",
+    },
+    validate,
+    onSubmit: (values) => {
+      const movieObj = convertToMovieObject(values);
+
+      if (values.dialogType === "Add Movie") {
```

Comment:
No magic strings - move "Add Movie" to const

Reply:
all done.